splitio/storage/inmemmory.py

```diff
@@ -419,3 +452,263 @@ def clear(self):
         """
         with self._lock:
             self._events = queue.Queue(maxsize=self._queue_size)
+
+class InMemoryTelemetryStorage(TelemetryStorage):
+    """In-memory telemetry storage."""
+
+    def __init__(self):
+        """Constructor"""
+        self._counters = {'iQ': 0, 'iDe': 0, 'iDr': 0, 'eQ': 0, 'eD': 0, 'sL': 0,
+                        'aR': 0, 'tR': 0}
+        self._latencies = {'mL': {'t': [], 'ts': [], 'tc': [], 'tcs': [], 'tr': []},
+                           'hL': {'sp': [], 'se': [], 'ms': [], 'im': [], 'ic': [], 'ev': [], 'te': [], 'to': []}}
+        self._exceptions = {'mE': {'t': 0, 'ts': 0, 'tc': 0, 'tcs': 0, 'tr': 0}}
+        self._records = {'IS': {'sp': 0, 'se': 0, 'ms': 0, 'im': 0, 'ic': 0, 'ev': 0, 'te': 0, 'to': 0},
+                         'sL': 0}
+        self._http_errors = {'sp': {}, 'se': {}, 'ms': {}, 'im': {}, 'ic': {}, 'ev': {}, 'te': {}, 'to': {}}
+        self._streaming_events = []
+        self._tags = []
+        self._integrations = {}
+        self._config = {'bT':0, 'nR':0, 'uC': 0}
+        self._map_latencies = {'Treatment': 't', 'Treatments': 'ts', 'TreatmentWithConfig': 'tc', 'TreatmentsWithConfig': 'tcs', 'Track': 'tr'}
+        self._lock = threading.RLock()
+
+    def record_config(self, config):
+        """Record configurations."""
+        with self._lock:
+            self._config['oM'] = self._get_operation_mode(config['operationMode'])
+            self._config['st'] = self._get_storage_type(config['operationMode'])
+            self._config['sE'] = config['streamingEnabled']
+            self._config['rR'] = self._get_refresh_rates(config)
+            self._config['uO'] = self._get_url_overrides(config)
+            self._config['iQ'] = config['impressionsQueueSize']
+            self._config['eQ'] = config['eventsQueueSize']
+            self._config['iM'] = self._get_impressions_mode(config['impressionsMode'])
+            self._config['iL'] = True if config['impressionListener'] is not None else False
+            self._config['hp'] = self._check_if_proxy_detected()
+            self._config['aF'] = config['activeFactoryCount']
+            self._config['rF'] = config['redundantFactoryCount']
```

Comment:
is this the same confir that the user passes when isntantiating the factory? if so, i'd suggest not to modify it with extra keys, but rather pass any extra data as seprate arguments (or another dict with contet info)